### ISSUE TITLE
TextEncoding polyfill in browser

### ISSUE THREAD

**Issue by vrachieru** [stomp-js/stompjs#133]
Using `v5.3.0` of this client in a `typescript` project that's ultimately compiled and then served by a `nginx` server raises an issue with the `TextEncoding\TextDecoding` polyfill as they do not work/are not present in the browser.  
This issue is present in both `IE11` and `MS Edge`.  

I am not actually a FE dev but I suspect the fix is making use of the ployfill in `spec/config/browser-config.js` as it is done with `node-config.js`.

```js
if (typeof TextEncoder !== 'function') {
  const TextEncodingPolyfill = require('text-encoding');
  TextEncoder = TextEncodingPolyfill.TextEncoder;
  TextDecoder = TextEncodingPolyfill.TextDecoder;
}
```
> Looking closer, it seems `spec` is just for tests and not actual runtime.

**Comment by kum-deepak** [comment]
You are correct MS browsers do need these. Please see https://stomp-js.github.io/guide/stompjs/rx-stomp/ng2-stompjs/2018/06/29/pollyfils-for-stompjs-v5.html

**Comment by vrachieru** [comment]
Thank you for the prompt reply @kum-deepak !

So basically someone using `stompjs` has to add polyfilling on thier own to the app.
Including the following in said app should suffice
```html
<script src="https://cdn.jsdelivr.net/npm/text-encoding@0.6.4/lib/encoding.min.js"></script>
```
as the presence of `TextEncoder` and `TextDecoder` is checked upon initialization.
```js
  if (!global['TextEncoder'])
    global['TextEncoder'] = TextEncoder;
  if (!global['TextDecoder'])
    global['TextDecoder'] = TextDecoder;

  if (typeof module !== "undefined" && module.exports) {
    module.exports = {
      TextEncoder: global['TextEncoder'],
      TextDecoder: global['TextDecoder'],
      EncodingIndexes: global["encoding-indexes"]
    };
  }
```

But wouldn't it be more straight forward to have some similar logic as in `node-config` in the `stompjs` lib itself (maybe in `index.ts`) so the user doesn't have to add the polyfill by himself?

**Comment by kum-deepak** [comment]
My reasoning for not including any Polyfills in the library distribution is following:

- All the Polyfills (including TextEncoder/TextDecoder) are part of web standards. So, eventually these will get implemented by each of the target browsers/platforms. I use https://caniuse.com/#search=TextEncoder to periodically check current support.
- A polyfill can be included targeting only the browsers that need them. Like https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/compatibility/ms537512(v=vs.85).
- Some environments like Angular have a common way to include polyfills. This allows the underlying webpack to, possibly, create alternate bundles for different browsers.

In summary, the strategy is to keep the default bundle small with ability to add pollyfils as needed. For example, always adding text-encoding to the stompjs bundle will double the size of the minified stompjs bundles.

**Comment by vrachieru** [comment]
I see. Thank you for the info @kum-deepak ! 🍻

**Comment by bdwenxi** [comment]
Mount TextEncoder and TextDecoder on global variables

```javascript
import {TextEncoder, TextDecoder} from 'text-encoding';

if (!global['TextEncoder']) {
    global['TextEncoder'] = TextEncoder;
}

if (!global['TextDecoder']) {
    global['TextDecoder'] = TextDecoder;
}
```
